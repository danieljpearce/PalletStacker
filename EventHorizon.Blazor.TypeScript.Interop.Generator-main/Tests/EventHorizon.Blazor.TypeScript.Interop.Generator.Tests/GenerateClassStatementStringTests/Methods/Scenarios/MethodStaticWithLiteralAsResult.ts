declare module Static.Class {
    export class ExampleClass {
        static ObjectLiteralFirstUnionResult(): {
            value1?: number;
        } | string;
        static ObjectLiteralLastUnionResult(): string | {
            value1?: number;
        };
        getVertexBuffers(): Nullable<{
            [key: string]: VertexBuffer;
        }>;
    }
}
