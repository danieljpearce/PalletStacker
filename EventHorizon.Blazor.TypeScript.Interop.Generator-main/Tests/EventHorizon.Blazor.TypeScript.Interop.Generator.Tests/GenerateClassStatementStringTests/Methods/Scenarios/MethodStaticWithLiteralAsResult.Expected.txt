/// Generated - Do Not Edit
namespace Static.Class
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods
        public static CachedEntity ObjectLiteralFirstUnionResult()
        {
            return EventHorizonBlazorInterop.Func<CachedEntity>(
                new object[]
                {
                    new string[] { "Static", "Class", "ExampleClass", "ObjectLiteralFirstUnionResult" }
                }
            );
        }

        public static string ObjectLiteralLastUnionResult()
        {
            return EventHorizonBlazorInterop.Func<string>(
                new object[]
                {
                    new string[] { "Static", "Class", "ExampleClass", "ObjectLiteralLastUnionResult" }
                }
            );
        }
        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods
        public CachedEntity getVertexBuffers()
        {
            return EventHorizonBlazorInterop.Func<CachedEntity>(
                new object[]
                {
                    new string[] { this.___guid, "getVertexBuffers" }
                }
            );
        }
        #endregion
    }
}